syntax = "proto3";

package smalltalk;

// Сервис для взаимодействия между хостами
service SmallTalkService {
  // Запрос задачи
  rpc GetTask (TaskRequest) returns (TaskResponse) {}
  // Сообщение о завершении задачи
  rpc CompleteTask (CompleteRequest) returns (CompleteResponse) {}
  // Проверка соединения
  rpc Ping (PingRequest) returns (PingResponse) {}
}

// Запрос задачи
message TaskRequest {
  int32 device_id = 1;
  int64 key_count = 2;
}

// Ответ с задачей
message TaskResponse {
  string start = 1;
  string end = 2;
  int32 col_idx = 3;  // Номер коллекции
}

// Запрос завершения задачи
message CompleteRequest {
  int32 device_id = 1;
  string start = 2;
  string end = 3;
  int32 col_idx = 4;  // Добавим col_idx для точного сохранения
}

// Ответ о завершении
message CompleteResponse {
  string status = 1;
}

// Запрос проверки соединения
message PingRequest {
  string message = 1;
}

// Ответ на проверку соединения
message PingResponse {
  string message = 1;
}